### ISSUE TITLE
cloudsmith: missing support for raw packages.

### ISSUE THREAD

**Issue by leamas** [mauroc/squiddio_pi#89]
Looking at the new stable cloudsmith repo at https://cloudsmith.io/~mauro-calvi/repos/squiddio-stable/packages it's not possible to download anything. The green "download" icon has   tooltip saying *Support for the Raw package backend is not currently enabled for this repository*.

Seems like some setting. Comparing with https://cloudsmith.io/~alec-leamas/repos/opencpn-plugins-stable/packages, the latter has the downloads enabled. I also note that the working setup has a open-source license defined for the repo.  Possibly that, or some other setting I guess.

**Comment by rgleason** [comment]
Mauro, in the repository 
https://cloudsmith.io/~mauro-calvi/repos/squiddio-stable/packages/
Light blue button to the upper right  "Set me up" is a drop down.  Pick "Raw" at the very bottom.
I did that and things started working better.

**Comment by rgleason** [comment]
Mauro, is this resolved yet? Can this be closed?

**Comment by leamas** [comment]
No, it's not.  It is still not possible to download from said repo.

**Comment by rgleason** [comment]
I just had a talk with Cloudsmith support. They're very helpful.

07:59 PM | Rick Gleason: Dear Lee, new to this. How do I enable a raw file download from a public repository that has had Setup Raw selected?

08:03 PM | Lee from Cloudsmith: Hey Rick! Is this for an open-source project?

08:04 PM | Rick Gleason: Yes

08:05 PM | Lee from Cloudsmith: Great. You just need to make sure you create an open-source repository when you're creating the repository. Otherwise "raw" package files are only supported on paid plans. That's why you can't download any files from the public repository (if you're currently not paying).

08:07 PM | Rick Gleason: I believe I created an opensource public repository, and under "Setup" button I selected "Raw"

08:07 PM | Lee from Cloudsmith: The reason it is like this, is to prevent users who aren't open-source or paying to abuse Cloudsmith as a file-sharing service. This isn't you, of course, but make sure you create the repositories as "open-source" with the right link to the open-source project. We're happy to support that for free.

08:07 PM | Rick Gleason: Does this mean Users cannot download?

08:08 PM | Lee from Cloudsmith: Only for "public" repositories. "open-source" repositories are still public, but free, with a greater allowance for storage/bandwidth.

08:08 PM | Lee from Cloudsmith: Let me just check to see what you've got.

08:08 PM | Rick Gleason: Ok possibly I did something wrong when setting up the repository?

08:08 PM | Rick Gleason: One is https://cloudsmith.io/~rick-gleason/repos/squiddio-pi/

08:10 PM | Lee from Cloudsmith: Yes, it looks like these are "public" rather than "open-source". Let me just grab a screenshot for you to show you what it looks like during setup.

![open_source_repository](https://user-images.githubusercontent.com/2363959/68076807-0b681800-fd90-11e9-9354-1d28ddff6c8b.png)

08:11 PM | Lee from Cloudsmith: [Image "open_source_repository.png"]

08:11 PM | Lee from Cloudsmith: You'll need to delete the existing repositories and re-create them to make them "open-source" though!

08:12 PM | Lee from Cloudsmith: By the way, you can also create an organization account if you'd like to manage the repositories with other users. If not, creating it all on your user is fine too. :-)

Alec, I will create a new item for this last topic, as per your preference.

**Comment by rgleason** [comment]
Dear Mauro, 
Hope all is going well.  

Success! The recent PR Merge from Alec resulted in builds and deployment to each of the directories:
[https://cloudsmith.io/~mauro-calvi/repos/](https://cloudsmith.io/~mauro-calvi/repos/)
mauro-calvi/manual <--Unlocked icon, just a Public Directory.
mauro-calvi/squiddio-pi <--Has a Heart Icon, hover over it "Open Source Directory"
mauro-calvi/squiddio-stable  <--Unlocked icon, just a Public Directory.

Your squiddio-pi directory does download because it has a heart and is opensource. 

The manual and squiddio-stable repositories are Public but not Opensource, so users and PI Installer will be unable to download from them.  Therefore for those directories to work for Opensource they  will need to be removed and recreated following the note above.  I've had to do this too and it takes a little while for the directories to be deleted, perhaps 10 minutes, then you can just recreate them selecting the heart for "Opensource" and completing the OpenSource License as GNU second version and the project url.

Best, Rick

**Comment by rgleason** [comment]
I think this can be closed now.

**Comment by rgleason** [comment]
I believe this can be closed now.
